src/cincinnati/mod.rs

```diff
@@ -130,6 +131,52 @@ impl Cincinnati {
     }
 }
 
+/// Write MOTD indicating a dead-end release, with the passed `reason`.
+/// If `reason` is `None`, the MOTD written is left as an empty file.
+fn refresh_deadend_motd(reason: Option<String>) -> Fallible<()> {
+    // Avoid showing partially-written messages using tempfile and
+    // persist (rename).
+    let mut f = tempfile::Builder::new()
+        .prefix(".deadend.")
+        .suffix(".motd.partial")
+        // Create the tempfile in the same directory as the final MOTD,
+        // to ensure proper SELinux labels are applied to the tempfile
+        // before renaming.
+        .tempfile_in("/run/zincati/public/motd.d")
```

Comment:
Just double-checking: is this one created with `0664 - zincati/zincati` permission too?

Reply:
tempfile creates the file with `0600` - had seen this but didn't think too much of it. Having the tmpfiles.d give it `0664` only to be replaced by a file with mode `0600` is confusing though - I updated this to set the mode of the tempfile to `0664`.

Reply:
Ack. Upstream library has this one open since some time: https://github.com/Stebalien/tempfile/issues/114.